PhotoStream/app/src/main/java/com/example/siyer/photostream/SearchFragment.java

```diff
@@ -0,0 +1,137 @@
+            @Override
+            public void onClick(View v) {
+                if (urlList.size()>0) {
+                    if (urlIndex < urlList.size() - 1) {
+                        urlIndex += 1;
+                    } else {
+                        urlIndex = 0;
+                    }
+                    getImageFromURL();
+                }
+            }
+        });
+        //OnCLick Listener for PREV button
+        final View prev_but = rootview.findViewById(R.id.prev_button);
+        prev_but.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (urlList.size() > 0) {
+                    if (urlIndex >= 1) {
+                        urlIndex -= 1;
+                    } else {
+                        urlIndex = urlList.size() - 1;
+                    }
+                    getImageFromURL();
+                }
+            }
+        });
+        //OnCLick Listener for ADD button
+        final View add_but = rootview.findViewById(R.id.add_button);
+        add_but.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (!collectionURL.contains(urlList.get(urlIndex))) {
+                    mDbHelper = new CollDbHelper(getActivity().getBaseContext()); //initialize database
+                    mDbHelper.writeDatabase(urlList.get(urlIndex));
+
+                    collectionURL.add(urlList.get(urlIndex));//Add it to collection
+                    Toast.makeText(getActivity().getBaseContext(), "Added", Toast.LENGTH_SHORT).show();
+                } else {
+                    Toast.makeText(getActivity().getBaseContext(), "Already in the collection", Toast.LENGTH_LONG).show();
+                }
+            }
+        });
+        //OnCLick Listener for changing into COLLECTION fragment
+        final View coll_but = rootview.findViewById(R.id.coll_button);
+        coll_but.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ((MainActivity) getActivity()).changeToColl();
+            }
+        });
+        //Initializing WebView
+        webImgView = (WebView) rootview.findViewById(R.id.webView);
+        webImgView.loadUrl(getString(R.string.default_img));
+        return rootview;
+    }
+
+    //Handles updating the WebView
+    public void getImageFromURL() {
+        String urlToShow = urlList.get(urlIndex);
+        webImgView.setWebViewClient(new WebViewClient());
+        WebSettings settings = webImgView.getSettings();
+        settings.setUseWideViewPort(true);
+        settings.setLoadWithOverviewMode(true);
+        
+        try {
+            webImgView.loadUrl(urlToShow);
+            Log.i("PRINTER", urlToShow);
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            Log.i("PRINTER", "WebView failed..." + "\t" + urlToShow);
```

Comment:
Adding some indication that the loading failed to the loser could be useful as a UI decision.